File: admin/src/services/team.go
package services

import "admin/models"

func CreateTeam(name string, creatorId string, gameID string) (string, error) {
	// チームを作成
	team,err := models.CreateTeam(name, gameID, creatorId)

	// エラー処理
	if err != nil {
		return "", err
	}

	return team.TeamID, nil
}

func DeleteTeam(teamid string) error {
	// チームを取得
	team, err := models.GetTeam(teamid)

	// エラー処理
	if err != nil {
		return err
	}
	
	// チームを削除
	err = team.Delete()

	// エラー処理
	if err != nil {
		return err
	}

	return nil
}

// ゲームに所属するチームを取得
func ListTeam(gameid string) ([]models.Team, error) {
	// ゲーム取得
	game, err := models.GetGame(gameid)

	// エラー処理
	if err != nil {
		return nil, err
	}

	// チームを取得
	teams, err := game.GetTeams()

	// エラー処理
	if err != nil {
		return nil, err
	}

	return teams, nil
}

func UpdateNickName(teamid string, name string) (models.Team, error) {
	// チームを取得
	team, err := models.GetTeam(teamid)

	// エラー処理
	if err != nil {
		return models.Team{}, err
	}

	// チームを更新
	err = team.UpdateNickName(name)

	// エラー処理
	if err != nil {
		return models.Team{}, err
	}

	return team, nil
}

File: nginx/statics/dev/scripts/team.js
// ゲームのチームを取得
async function ListTeam(gameid) {
    // トークンを取得
    const token = await GetToken();

    // リクエストを送る
    const req = await fetch("/admin/team/list", {
        method: "GET",
        headers: {
            "Content-Type": "application/json",
            "Authorization": token,
            "gameid": gameid
        },
    });

    return req.json();
}

// チームリストを更新
async function RefreshTeam(gameid) {
    // チームリストをクリア
    teamlist.innerHTML = "";

    const teams = await ListTeam(gameid);
    console.log(teams["msg"]);

    teams["msg"].forEach((team) => {
        ShowTeam(team["TeamID"], team["Name"], team["Creator"], team["Status"], team["NickName"]);
    });
}

// チームを削除する
async function DeleteTeam(teamid) {
    // トークンを取得
    const token = await GetToken();

    // リクエストを送る
    const req = await fetch("/admin/team/delete", {
        method: "DELETE",
        headers: {
            "Content-Type": "application/json",
            "Authorization": token,
        },
        body: JSON.stringify({ "teamid": teamid }),
    });

    return req.json();
}


// 関数をまとめた処理
async function CreateTeam(name,gameid) {
    // トークンを取得
    const token = await GetToken();

    // リクエストを送る
    const req = await fetch("/admin/team/create", {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
            "Authorization": token,
        },
        body: JSON.stringify({ 
            "name": name,
            "gameid": gameid
        }),
    });

    return req.json();
}

// リンク解除
async function RemoveLink(teamid) {
    // トークンを取得
    const token = await GetToken();

    // リクエストを送る
    const req = await fetch("/admin/team/unlink", {
        method: "DELETE",
        headers: {
            "Content-Type": "application/json",
            "Authorization": token,
        },
        body: JSON.stringify({ "teamid": teamid }),
    });

    return req.json();
}
